bug_report_page.php

```diff
@@ -605,7 +605,22 @@
 			<label for="attach_tag"><?php echo lang_get( 'tag_attach_long' ) ?></label>
 		</th>
 		<td>
-			<?php print_tag_input( '' ); ?>
+			<?php
+				if( $f_master_bug_id > 0 ) {
+					# pre-fill tag string when cloning from master bug
+					$t_tags = [];
+					foreach( tag_bug_get_attached( $f_master_bug_id ) as $t_tag ) {
+						array_push( $t_tags, $t_tag["name"] );
+					}
+					$t_tag_string = implode(
+						config_get( 'tag_separator' ), $t_tags
+					);
+					print_tag_input( $f_master_bug_id, $t_tag_string );
```

Comment:
While testing this, I noticed that there is actually a bug in *tag_get_candidates_for_bug()*. The function returns all tags, including those already attached to the specified bug, in contradiction with the function's documentation.

Fixing this issue, the dropdown list of tags no longer contains those associated with the master bug id, and they are added to the text box (as expected).

The problem is that if the user changes the text box's contents (e.g. editing or removing a tag), then they are no longer able to add the original tag back by selecting it from the list.

Therefore I wonder if it would not be better to call `print_tag_input( 0, $t_tag_string );` in this context.

Thoughts ?

Reply:
Hm, but wouldn't it mean all of the tags are printed into the text input? If so, then such behavior is undesired because the point of cloning is to have the same tags as the master issue, not all of the tags Mantis has. In small instances that might be even fine, however imagine cloning an issue in mantisbt instance... that might be quite a trouble :)

Reply:
> wouldn't it mean all of the tags are printed into the text input

No, the list of tags included in the text box are specified with the 2nd parameter ($t_tag_string).

Reply:
Ok, let me patch it then, hopefully it helps with the bug you mentioned.

Reply:
Thanks